examples/fully-featured/flake.nix

```diff
@@ -134,7 +138,22 @@
       devShellBuilder = channels: channels.nixpkgs.mkShell { name = "devShell"; };
 
       # Evaluates to `checks.<system>.attributeKey = "attributeValue"`
-      checksBuilder = channels: { check = channels.nixpkgs.runCommandNoCC "test" { } "echo test > $out"; };
+      checksBuilder = channels:
+        let
+          booleanCheck = cond:
+            if cond
+            then channels.nixpkgs.runCommandNoCC "success" { } "echo success > $out"
+            else channels.nixpkgs.runCommandNoCC "failure" { } "exit 1";
+        in
+        {
+          check = channels.nixpkgs.runCommandNoCC "test" { } "echo test > $out";
+          # Modules (and lib) from patched nixpkgs are used
+          summerHasCustomModuleConfigured = booleanCheck (self.nixosConfigurations.Summer.config.patchedModule.test == "test");
+          # nixpkgs config from host-specific module is used
+          summerHasPackageOverridesConfigured = booleanCheck (self.nixosConfigurations.Summer.config.nixpkgs.pkgs.config ? packageOverrides);
+          # nixpkgs config from channel is also used
+          summerHasUnfreeConfigured = booleanCheck (self.nixosConfigurations.Summer.config.nixpkgs.pkgs.config ? allowUnfree);
+        };
```

Comment:
Alright, you put more work into it than expected - I would have just set some host-specific configuration and checked if it builds.

I like the fact that it's a check but I am not sure if this should be part of the example. But then again - this example is already pretty overwhelming.

@Pacman99 @blaggacao Any thoughts?

Reply:
IMO, it's completely ok to have the fulky featured example overflow. Where can we have such detailed documentation that is not called "tests", otherwise? :wink: